jerry-debugger/jerry-client-ws.py

```diff
@@ -0,0 +1,676 @@
+JERRY_DEBUGGER_RESOURCE_NAME = 7
+JERRY_DEBUGGER_RESOURCE_NAME_END = 8
+JERRY_DEBUGGER_FUNCTION_NAME = 9
+JERRY_DEBUGGER_FUNCTION_NAME_END = 10
+JERRY_DEBUGGER_RELEASE_BYTE_CODE_CP = 11
+JERRY_DEBUGGER_BREAKPOINT_HIT = 12
+JERRY_DEBUGGER_BACKTRACE = 13
+JERRY_DEBUGGER_BACKTRACE_END = 14
+
+# Messages sent by the client to server
+JERRY_DEBUGGER_FREE_BYTE_CODE_CP = 1
+JERRY_DEBUGGER_UPDATE_BREAKPOINT = 2
+JERRY_DEBUGGER_STOP = 3
+JERRY_DEBUGGER_CONTINUE = 4
+JERRY_DEBUGGER_STEP = 5
+JERRY_DEBUGGER_NEXT = 6
+JERRY_DEBUGGER_GET_BACKTRACE = 7
+
+PORT = 5001
+MAX_BUFFER_SIZE = 128
+HOST = "localhost"
+
+
+def arguments_parse():
+    parser = argparse.ArgumentParser(description='JerryScript debugger client')
+
+    parser.add_argument('address', action='store', nargs='?', default='localhost:5001', help='specified a custom network setup (default: %(default)s)')
+    parser.add_argument('-v', '--verbose', action='store_true', default=False, help='increase verbosity (default: %(default)s)')
+
+    args = parser.parse_args()
+
+    if ':' not in args.address:
+        print('Wrong address settings: use the "IP:PORT" format.')
+    else:
+        unique_host, unique_port = args.address.split(':')
+        HOST = unique_host
+        PORT = unique_port
+        print('Address setup: %s:%s' % (unique_host, unique_port))
+
+    if args.verbose:
+        logging.basicConfig(format='%(levelname)s: %(message)s' , level=logging.DEBUG)
+        logging.debug('Debug logging mode: ON')
+
+
+class JerryBreakpoint(object):
+
+    def __init__(self, line, offset, function):
+        self.line = line
+        self.offset = offset
+        self.function = function
+        self.active_index = -1
+
+    def to_string(self):
+        result = self.function.source
+
+        if result == '':
+            result = '<unknown>'
+
+        result += ":%d" % (self.line)
+
+        if self.function.name:
+            result += ' (in %s)' % (self.function.name)
+        return result
+
+    def __repr__(self):
+        return ("Breakpoint(line:%d, offset:%d, active_index:%d)"
+            % (self.line, self.offset, self.active_index))
+
+
+class JerryFunction(object):
+
+    def __init__(self, byte_code_cp, source, name, lines, offsets):
+        self.byte_code_cp = byte_code_cp
+        self.source = source
+        self.name = name
+        self.lines = {}
+        self.offsets = {}
+        self.first_line = -1
+
+        if len(lines) > 0:
+            self.first_line = lines[0]
+
+        for i in range(len(lines)):
+            line = lines[i]
+            offset = offsets[i]
+            breakpoint = JerryBreakpoint(line, offset, self)
+            self.lines[line] = breakpoint
+            self.offsets[offset] = breakpoint
+
+    def __repr__(self):
+        result = ('Function(byte_code_cp:0x%x, source:\'%s\', name:\'%s\', { '
+            % (self.byte_code_cp, self.source, self.name))
+
+        comma_needed = False
+        for breakpoint in self.lines.values():
```

Comment:
If I understand correctly you just simply join the breakpoints as strings with a separator together. If so this would be better imho:
```
breakpoints = ','.join([str(breakpoint) for breakpoint in self.lines.values()])
```
and then you can add the breakpoints into the format string above